src/ai.c

```diff
@@ -2132,6 +2134,45 @@ static int aiL_getnearestplanet( lua_State *L )
 }
 
 
+/**
+ * @brief Get the nearest friendly planet to a given position.
+ *
+ *    @luatparam vec2 pos Position close to the planet.
+ *    @luatreturn Planet|nil
+ *    @luafunc planetfrompos
+ */
+static int aiL_getplanetfrompos( lua_State *L )
+{
+   Vector2d *pos;
+   double dist, d;
+   int i, j;
+   LuaPlanet planet;
+
+   pos = luaL_checkvector(L,1);
+
+   /* cycle through planets */
+   for (dist=HUGE_VAL, j=-1, i=0; i<array_size(cur_system->planets); i++) {
```

Comment:
Do you mean `math.huge`?

Reply:
I don't know. I simply copy/pasted the previous function :/